packages/server/src/script/resendSignupEmails.ts

```diff
@@ -0,0 +1,136 @@
+    for (const info of registrations) {
+        // Setup.
+        const method = toContactMethod(info.contact)
+        if (method == null) {
+            console.log('No contact method found for '.concat(info.id))
+            continue
+        }
+        const resendReasonAndOriginalSignupDate = resendReason
+            .concat(" Voter originally signed up for VoteByMail.io on ")
+            .concat(info.created.toDate().toISOString().split('T')[0])
+        const letter = new Letter(info, method, info.id, resendReasonAndOriginalSignupDate)
+        const pdfBuffer = await toPdfBuffer(letter.render('html'), await letter.form)
+
+        // Send emails.
+        const mgResponse = await sendSignupEmail(
+            letter,
+            info.email,
+            method.emails,
+            { pdfBuffer },
+            )
+        if (!mgResponse || mgResponse.message != 'Queued. Thank you.') {
+            console.log('Error sending email for ' + info.id + '. Exiting.')
+            console.log(mgResponse)
+            return
+        }
+        console.log('Successfully sent emails for ' + info.id)
+
+
+        // Number of resends.  _0 is the first, _1 the second, etc.
+        const numResends = info.resends ? info.resends.length : 0
+        const file = storageFileFromId(info.id.concat('_') + numResends)
+        await file.upload(pdfBuffer)
+
+        // Send faxes
+        let twilioResponses: TwilioResponse[] = []
+        if (method.faxes.length > 0) {
+            const uri = await file.getSignedUrl(24 * 60 * 60 * 1000)
+            const resposnes = await sendFaxes(uri, method.faxes)
+            twilioResponses = resposnes.map(({url, sid, status}) => ({url, sid, status}))
+
+            for (const twilioResponse of twilioResponses) {
+                if (twilioResponse.status != "queued") {
+                    console.log('Error sending faxes for ' + info.id + '. Exiting.')
+                    console.log(twilioResponse)
+                    return
+                }
+            }
+          }
+
+        // Update voter data in firestore.
+        const resend = { dateTime: new Date().getTime(), reason: resendReason }
+        await firestoreService
+            .db
+            .collection('StateInfo')
+            .doc(info.id)
+            .update({
+                resends: admin.firestore.FieldValue.arrayUnion(resend)
+            })
+    }
+}
+
+// Reads './data/conditions.json' and resends emails to all voter sign ups that satisfy conditions.
+const main = async () => {
+    const conditions = input['conditions']
+    // Ensure conditions in input JSON file has at least one entry.
+    if (Object.keys(conditions).length < 1) {
+        console.log('Must have at least one condition specified.')
+        return
+    }
+
+    // Fetch registrations.
+    let query = firestoreService.db.collection('StateInfo').where(conditions[0].field, conditions[0].op as WhereFilterOp, conditions[0].value)
```

Comment:
Again, I would just write the full query directly in the file and import the right one here.  You can then save off multiple queries and import the right one to run.